src/main/java/org/sigmah/server/security/impl/AccessRights.java

```diff
@@ -22,413 +22,363 @@
+		sperm(commandToken(GetReminders.class), GrantType.AUTHENTICATED_ONLY);
+		sperm(commandToken(GetUsersByOrganization.class), GrantType.AUTHENTICATED_ONLY);
+		sperm(commandToken(GetUsers.class), GrantType.AUTHENTICATED_ONLY);
+		sperm(commandToken(GetUsersWithProfiles.class), GrantType.AUTHENTICATED_ONLY);
+		sperm(commandToken(GetUsersByOrgUnit.class), GrantType.AUTHENTICATED_ONLY);
+		sperm(commandToken(GetValue.class), GrantType.AUTHENTICATED_ONLY);
+		sperm(commandToken(Synchronize.class), GrantType.AUTHENTICATED_ONLY);
+		sperm(commandToken(UpdateProject.class), GrantType.AUTHENTICATED_ONLY);
+		sperm(commandToken(UpdateProjectFavorite.class), GrantType.AUTHENTICATED_ONLY);
+		sperm(commandToken(UpdateProjectTeamMembers.class), GrantType.AUTHENTICATED_ONLY, GlobalPermissionEnum.EDIT_PROJECT_TEAM_MEMBERS);
+		sperm(commandToken(UploadSlice.class), GrantType.AUTHENTICATED_ONLY);
+
+		// Servlet methods.
+		sperm(servletToken(Servlet.FILE, ServletMethod.DOWNLOAD_LOGO), GrantType.AUTHENTICATED_ONLY);
+		sperm(servletToken(Servlet.FILE, ServletMethod.DOWNLOAD_FILE), GrantType.AUTHENTICATED_ONLY);
+		sperm(servletToken(Servlet.FILE, ServletMethod.DOWNLOAD_ARCHIVE), GrantType.AUTHENTICATED_ONLY);
+	}
+
+	/**
+	 * Granted tokens that are always granted in order to optimize application processes.
+	 */
+	static {
+		grantedTokens.add(commandToken(SecureNavigationCommand.class));
+	}
+
+	/**
+	 * Grants or refuse {@code user} access to the given {@code token}.
+	 *
+	 * @param user
+	 *          The user (authenticated or anonymous).
+	 * @param token
+	 *          The resource token (page, command, servlet method, etc.).
+	 * @param originPageToken
+	 *          The origin page token <em>(TODO Not used yet)</em>.
+	 * @param mapper
+	 *          The mapper service.
+	 * @return {@code true} if the user is granted, {@code false} otherwise.
+	 */
+	static boolean isGranted(final User user, final String token, final String originPageToken, final Mapper mapper) {
+
+		if (grantedTokens.contains(token)) {
+			// Granted tokens ; avoids profile aggregation if user is authenticated.
+			return true;
+		}
+
+		if (!permissions.containsKey(token)) {
+			if (LOG.isWarnEnabled()) {
+				LOG.warn("No security permission can be found for token '{}'. Did you forget to declare corresponding 'sperm'?", token);
+			}
+			return isGranted(user, MISSING_TOKEN, originPageToken, mapper);
+		}
+
+		final Pair<GrantType, Set<GlobalPermissionEnum>> grantData = permissions.get(token);
+		final GrantType grantType = grantData.left;
+
+		if (user == null || ServletExecutionContext.ANONYMOUS_USER.equals(user)) {
+			// Anonymous user.
+			return grantType != null && grantType != GrantType.AUTHENTICATED_ONLY;
+		} else {
+			// Authenticated user.
+			if (grantType != null && grantType == GrantType.ANONYMOUS_ONLY) {
+				return false;
+			} else {
+				for (Map.Entry<Integer, ProfileDTO> aggregatedProfileEntry : Handlers.aggregateProfiles(user, mapper).entrySet()) {
```

Comment:
Instead of `entrySet()`, `values()` should be called since the key of the entry is not needed in the body of the loop.